src/vmspawn/vmspawn-util.c

```diff
@@ -0,0 +1,341 @@
+                if (r == -ENOMEM)
+                        return r;
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to read contents of %s/%s - ignoring: %m", JSON_PATH(i));
+                        continue;
+                }
+
+                /* if the file is empty we need to mask any later occurances of this file */
+                if (contents_sz == 0) {
+                        size_t j;
+                        for (j = 1; j < 3; j++) {
+                                if (i + j >= n_cfs)
+                                        break;
+
+                                if (!streq(files[i].name, files[i + j].name))
+                                        break;
+
+                                log_debug("Less specific file %s/%s masked by empty %s/%s", JSON_PATH(i + j), JSON_PATH(i));
+                        }
+                        i += j - 1;
+                        continue;
+                }
+
+                r = json_parse(contents, 0, &config_json, NULL, NULL);
+                if (r == -ENOMEM)
+                        return r;
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to parse the JSON in %s/%s - ignoring: %m", JSON_PATH(i));
+                        continue;
+                }
+
+                JsonDispatch table[] = {
+                        { "description",     JSON_VARIANT_STRING, NULL,               0,                                JSON_MANDATORY },
+                        { "interface-types", JSON_VARIANT_ARRAY,  NULL,               0,                                JSON_MANDATORY },
+                        { "mapping",         JSON_VARIANT_OBJECT, firmware_mapping,   0,                                JSON_MANDATORY },
+                        { "targets",         JSON_VARIANT_ARRAY,  NULL,               0,                                JSON_MANDATORY },
+                        { "features",        JSON_VARIANT_ARRAY,  json_dispatch_strv, offsetof(FirmwareData, features), JSON_MANDATORY },
+                        { "tags",            JSON_VARIANT_ARRAY,  NULL,               0,                                JSON_MANDATORY },
+                        {}
+                };
+
+                _cleanup_(firmware_data_freep) FirmwareData* fwd = firmware_data_new();
+                if (!fwd)
+                        return -ENOMEM;
+
+                fwd->firmware = strdup("WOW A HELLA LONG STRING");
+                r = json_dispatch(config_json, table, NULL, 0, fwd);
+                if (r == -ENOMEM)
+                        return r;
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to extract the required fields from the JSON in %s/%s - ignoring: %m", JSON_PATH(i));
+                        continue;
+                }
+
+                int sb_present = 0;
+                STRV_FOREACH(s, fwd->features)
+                        if (streq(*s, "secure-boot")) {
```

Comment:
strv_find()

Reply:
would it be appropriate to use !!strv_find(...) to turn the return into an int?

Reply:
Fixed in c053f06da5.